docs/template-variables.rst

```diff
@@ -0,0 +1,393 @@
+``tag``             str         Friendly category name
+==================  ==========  =====================================================
+
+Variables available in galleries
+--------------------------------
+
+.. class:: table table-bordered table-striped
+
+======================  ==========  ===============================================================================
+Name                    Type        Description
+======================  ==========  ===============================================================================
+``crumbs``              list        Breadcrumbs for this page
+``enable_comments``     bool        Whether or not comments are enabled in galleries
+``folders``             list        List of folders (contains *path, title* tuples)
+``permalink``           str         Permanent link to this page
+``photo_array``         list        Photo array (contains dicts with image data: url, url_thumb, title, size{w, h})
+``photo_array_json``    str         Photo array in JSON format
+``post``                Post?       The Post object for this gallery
+``thumbnail_size``      int         ``THUMBNAIL_SIZE`` setting
+======================  ==========  ===============================================================================
+
+
+Variables available in listings
+-------------------------------
+
+.. class:: table table-bordered table-striped
+
+==================  ==========  ========================================
+Name                Type        Description
+==================  ==========  ========================================
+``code``            str         Highlighted source code (HTML fragment)
+``crumbs``          list        Breadcrumbs for this page
+``folders``         list<str>   List of subfolders
+``files``           list<str>   List of files in the folder
+``source_link``     str         Link to the source file
+==================  ==========  ========================================
+
+Variables available in tag pages
+--------------------------------
+
+.. class:: table table-bordered table-striped
+
+=========  =====  =================
+Name       Type   Description
+=========  =====  =================
+``kind``   str    Always ``"tag"``
+``tag``    str    Tag name
+=========  =====  =================
+
+Variables available in the “Tags and categories” page (``tags.tmpl``)
+---------------------------------------------------------------------
+
+.. class:: table table-bordered table-striped
+
+==============  ======  ===========================================================================================================
+Name            Type    Description
+==============  ======  ===========================================================================================================
+``items``       list    Tags *(name, link)*
+``cat_items``   list    Categories *(name, full name, path, link, indent levels, indent to change before, indent to change after)*
```

Comment:
I think that looks good. There should be somewhere (either here or above where `<overview_page_hierarchy_variable_name>` and `<overview_page_hierarchy_variable_name + "_with_postcount">` are defined) an explanation what indent levels, indent to change before, and indent to change after means. (Or is that what you meant with "how should I phrase"?)

I would write something like:

The indenting information can be render the items as a tree. The values have the following meanings:
 * `indent levels` is a list of pairs `(current_i, count_i)` giving the current position (`0`, ..., `count_i-1`) and maximum (`count_i`) in the hierarchy level `i`;
 * `indent to change before` is the difference of hierarchy levels between the previous and the current item; positive values indicate that the current item is indented further in and can be used to open HTML tags before the item;
 * `indent to change after` is the difference of hierarchy levels between the current and the next item; negative values indicate that the current item is indented further in and can be used to close HTML tags after the item.

Example:
```
 +--- levels:[(0,3)], before:1, after:0
 +-+- levels:[(1,3)], before:0, after:1
 | +--- levels:[(1,3), (0,2)], before:1, after:0
 | +-+- levels:[(1,3), (1,2)], before:0, after:1
 |   +--- levels:[(1,3), (1,2), (0, 1)], before:1, after:-2
 +-+- levels:[(2,3)], before:-2, after:1
   +- levels:[(2,3), (0,1)], before:1, after:-2
```
See `tags.tmpl` in the base themes for examples on how to render a tree as nested unordered lists in HTML.